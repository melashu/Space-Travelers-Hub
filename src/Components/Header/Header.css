.header {
  display: flex;
  justify-content: space-between;
  margin: 1rem;
  align-items: center;
  box-shadow: 0 5px 5px -10px #111;
  padding: 0.3rem;
  color: black;
}

h1 {
  margin: 0 0 0 0.7rem;
  padding: 0;
}

.nav-links {
  display: flex;
  gap: 0.7rem;
  padding-right: 10%;
}

.nav-link {
  text-decoration: none;
  font-weight: 600;
  transition: 200ms;
  padding: 10%;
}

.link:hover {
  color: #02bab7;
}

.logo {
  display: flex;
}
