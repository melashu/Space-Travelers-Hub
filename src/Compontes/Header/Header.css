.header {
  display: flex;
  justify-content: space-between;
  padding: 8px;
  box-shadow: 1px 2px 5px #999;
  text-align: center;
}

.logo {
  display: flex;
  align-items: center;
}

.logo img {
  width: 80px;
  height: 80px;
}

.comapny-title {
  font-size: 28px;
  font-weight: bold;
}

.list {
  list-style: none;
  display: flex;
  gap: 20px;
  padding: 10px;
}

li {
  font-size: 40px;
}

li a {
  text-decoration: none;
  font-weight: 400;
  font-size: 25px;
}
