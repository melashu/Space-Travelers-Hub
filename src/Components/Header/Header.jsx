import React from 'react'
import Logo from './planet.png';
import { NavLink } from 'react-router-dom';
import './Header.css';

export default function Header() {
  return (
    <div className="header">
      <div className="logo">
        <img src={Logo} width={50} height={50} alt="planet-logo" />
        <h1>Space Traveler&apos; s Hub</h1>
      </div>
      <div className="nav-links">
        <NavLink to="/rockets" className="nav-link">Rockets</NavLink>
        <NavLink to="/missions" className="nav-link">Missions</NavLink>
        <NavLink to="/profile" className="nav-link">Profile</NavLink>
      </div>
    </div>
  );
}