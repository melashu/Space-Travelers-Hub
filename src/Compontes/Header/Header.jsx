import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import './Header.css';
import logo from '../../images/logo.png';

const Header = () => {
  const activeStyle = {
    textDecoration: 'underline',
  };
  return (
    <header className="header">
      <Link to="/" style={{ textDecoration: 'none' }}>
        <div className="logo">
          <img src={logo} alt="" />
          <span className="comapny-title"> Space-Travelers-Hub</span>
        </div>
      </Link>
      <ul className="list">
        <li>
          {' '}
          <NavLink
            to="/rocket"
            style={({ isActive }) => (isActive ? activeStyle : undefined)}
          >
            Rocket
          </NavLink>
        </li>
        <li>
          {' '}
          <NavLink
            to="/mission"
            style={({ isActive }) => (isActive ? activeStyle : undefined)}
          >
            Mission
          </NavLink>
        </li>
        <li>
          {' '}
          <NavLink
            to="/profile"
            style={({ isActive }) => (isActive ? activeStyle : undefined)}
          >
            Profile
          </NavLink>
        </li>
      </ul>
    </header>
  );
};

export default Header;
