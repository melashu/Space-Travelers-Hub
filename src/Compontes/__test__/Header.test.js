import "@testing-library/jest-dom";
import Header from "../../Compontes/Header/Header";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";

const header = () => {
  <Provider store={store}>
    <Header />
  </Provider>;
};

describe("Should Header render correctly", () => {
  it("Test Header Componenet", () => {
    const component = renderer.create(<header />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
