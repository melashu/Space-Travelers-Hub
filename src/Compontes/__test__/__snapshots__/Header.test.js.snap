// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Header render correctly Test Header Componenet 1`] = `<header />`;

exports[`Should Header render correctly Test Mission Componenet 1`] = `<header />`;

exports[`Should mission render correctly Test Mission Componenet 1`] = `<mission />`;
